WordPress/Classes/Services/SharingService.swift

```diff
@@ -106,50 +130,72 @@ open class SharingService: LocalCoreDataService {
+        let blogObjectID = blog.objectID
+        coreDataStack.performAndSave({ context -> PubConnUpdateResult in
+            let blogInContext = try context.existingObject(with: blogObjectID) as! Blog
+            let pubConnInContext = try context.existingObject(with: pubConn.objectID) as! PublicizeConnection
+            let oldValue = pubConnInContext.shared
+            pubConnInContext.shared = shared
+            return (
+                oldValue: oldValue,
+                siteID: pubConnInContext.siteID,
+                connectionID: pubConnInContext.connectionID,
+                service: pubConnInContext.service,
+                remote: self.remoteForBlog(blogInContext)
+            )
+        }, completion: { result in
+            switch result {
+            case let .success(value):
+                if value.oldValue == shared {
+                    success?()
+                    return
+                }
 
-            let blogObjectID = blog.objectID
-            let siteID = pubConn.siteID
-            guard let remote = remoteForBlog(blog) else {
-                return
-            }
-            remote.updatePublicizeConnectionWithID(pubConn.connectionID,
-                shared: shared,
-                forSite: siteID,
-                success: { remoteConnection in
-                    let properties = [
-                        "service": pubConn.service,
-                        "is_site_wide": NSNumber(value: shared).stringValue
-                    ]
-                    WPAppAnalytics.track(.sharingPublicizeConnectionAvailableToAllChanged, withProperties: properties, withBlogID: siteID)
-                    do {
-                        _ = try self.createOrReplacePublicizeConnectionForBlogWithObjectID(blogObjectID, remoteConnection: remoteConnection)
-                        ContextManager.sharedInstance().save(self.managedObjectContext, completion: success, on: .main)
-                    } catch let error as NSError {
-                        DDLogError("Error creating publicize connection from remote: \(error)")
-                        failure?(error)
+                value.remote?.updatePublicizeConnectionWithID(
+                    value.connectionID,
+                    shared: shared,
+                    forSite: value.siteID,
+                    success: { remoteConnection in
+                        let properties = [
+                            "service": value.service,
+                            "is_site_wide": NSNumber(value: shared).stringValue
+                        ]
+                        WPAppAnalytics.track(.sharingPublicizeConnectionAvailableToAllChanged, withProperties: properties, withBlogID: value.siteID)
+
+                        self.coreDataStack.performAndSave({ context in
+                            _ = try self.createOrReplacePublicizeConnectionForBlogWithObjectID(blogObjectID, remoteConnection: remoteConnection, in: context)
```

Comment:
Would it be useful to make the method `@discaradableResult`? Or is the fact that we are calling it and ignoring it's result a warning bell?

Reply:
Update: There is code below that calls the method without an assignment, so I guess this was just something lost in translation here.

```suggestion
                            try self.createOrReplacePublicizeConnectionForBlogWithObjectID(blogObjectID, remoteConnection: remoteConnection, in: context)
```

Reply:
Updated in https://github.com/wordpress-mobile/WordPress-iOS/pull/20265/commits/eaea3de012e00698fbf13dcdb55eba584305d14a